### ISSUE TITLE
examples files un-runnable

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#5037]
Maybe run these in the CI?  We shouldn't be advertising broken examples.

```
$ python examples/run_all.py
Are you sure you want to run all of the examples?
          This is done mainly to check that they are up to date.
          (y/n) >>> y


Executing example file example_glsar.py
---------------------------------------
**************************************
ERROR in example file example_glsar.py
**************************************
Traceback (most recent call last):
  File "examples/run_all.py", line 43, in <module>
    exec(open(run_all_f).read())
IOError: [Errno 2] No such file or directory: 'example_glsar.py'
```

**Comment by josef-pkt** [comment]
Where are they advertised?

**Comment by jbrockmendel** [comment]
Front page of the repo.

**Comment by josef-pkt** [comment]
I don't see a link or mentioning of those examples.

https://github.com/statsmodels/statsmodels/blob/master/examples/run_all.py  looks misplaced, most likely a left over file from the transition from python examples to notebooks

the real run_all is for the internal examples
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/examples/run_all.py

which runs with exceptions after skipping all the kernel regression and multivariate kde files.
(maybe parallel processing doesn't work when I run this in IDLE.

```
Modules that raised exception:
['example_discrete_mnl.py', 'example_kde.py', 'ex_emplike_1.py', 'ex_emplike_2.py', 'ex_generic_mle.py', 'ex_misc_tarma.py', 'ex_misc_tmodel.py', 'ex_pairwise.py', 'ex_pandas.py', 'ex_proportion.py', 'ex_regressionplots.py', 'ex_sandwich2.py', 'ex_scatter_ellipse.py', 'ex_shrink_pickle.py', 'try_gee.py', 'try_tukey_hsd.py']
```

those internal example files are very low priority, unless somebody wants to convert them to notebooks and make proper DOC examples out of them (which happened with several of the initial examples)

**Comment by josef-pkt** [comment]
For some time in the past I thought having dirty examples is better than having no examples.
Now we hide them on private computers or gists.

**Comment by jbrockmendel** [comment]
> Now we hide them on private computers or gists.

Huh?

**Comment by josef-pkt** [comment]
Whenever I need to check something I need examples.
Now they are just in the script and notebook folder on my computers, and a few in gists.
The same happens for many PRs where we often don't get any public examples.
(Nobody can complain that they might be out of date.)

The main source of publicly accessible examples that are maintained are the unit test cases.

**Comment by jbrockmendel** [comment]
> Now they are just in the script and notebook folder on my computers, and a few in gists.

That is a great place for files not useful to the public.

> [...] that are maintained [...]

the point of this Issue is that things that are _not_ maintained don't belong in the repo.  Or at the very least belong in the sandbox (#4887)

**Comment by josef-pkt** [comment]
(same old story)

As a maintainer and default fast response person I want to have as much information available as possible, even if it's outdated. It's very difficult to figure out what code is supposed to do if there is not enough of a supporting "paper trail".

(I spend some time going through most of the subpackages of scipy when scipy was around 0.6 or 0.7 without hints about whether I misinterpret the conceptual background that I didn't know much about, whether there were bugs, incomplete implementation or other weird things, and for many parts there was no maintainer to ask questions to. Only after many years of cleanup was the left-over code that didn't fit in scipy deleted. Having even old examples or even shaky unit tests would have helped me a lot. 
On the other hand, going through scipy was an interesting detective story for me. E.g. translate scipy.signal signal processing terminology into econometrics terminology and figure out whether the behavior that I didn't understand is a bug or a feature.)
